### ISSUE TITLE
Range select, more extensions and others

### ISSUE THREAD

**Issue by AbdulrahmanHadz** [inorichi/tachiyomi#1293]
Hi, i would like to request for the app to have chapter range select so that you can select from chapter 13-45 for example. I would also like to request an extension for lhscans.com for raw manga, chapter grid view and to have a resume button that lets you read for the chapter you were on last even if the chapter has been marked as read.

Thank you,
and keep up the hard work

**Comment by Flat** [comment]
Requests for additional sources do not belong in this repository. They go here https://github.com/inorichi/tachiyomi-extensions

Please elaborate on a range select. You can already long-tap to select a range of chapters.

A resume button already exists and should pick up where you left off.

What is the rationale of a grid view for chapters? What information would this include?

**Comment by Madidas** [comment]
> Please elaborate on a range select. You can already long-tap to select a range of chapters.

Long-tap allows user to select multiple chapters but selecting those chapters must be done one chapter at the time (unless it's changed). Lots of tapping and scrolling if you need to select lots of chapters (like 15+ chapters at once).
 
I guess @AbdulrahmanHadz would like to have feature to select two different chapters and also have every chapter between those two selected chapters to be also selected. Like extra option in the "three dot menu". Or just a text field/box/menu option to type where the selection of chapters starts and ends (Select everything from Chapter 13 to Chapter 45). Could be useful for manipulation of large set of chapters (set read/unread, download/delete certain chapters).

That is how I interpreted that "chapter range select" suggestion.

**Comment by AbdulrahmanHadz** [comment]
What i meant is what @Madidas explained, the continue reading only works with unread chapters, i want to be able to resume reading a chapter that i have already read before and the app has marked as read (e.g. re-reading a series). The resume button gives a float notification that next chapter is not found. 
The grid view would help when navigating a series that has a lot of chapters and you want to reach the chapter quicker than needing to scroll.
Thanks, and next time i'll put this thread in the correct repository.

**Comment by Flat** [comment]
Regarding resume button: https://github.com/inorichi/tachiyomi/issues/1186#issuecomment-366460272

Tl;dr working as intended, mark them un-read if you're re-reading. 

The range select is probably doable if someone has time to implement it.

As for the grid view, how would this be different than the list view? Do you mean just take the current list of chapters and cut the width in half to have two entries in one row? Images for each chapter?

**Comment by AbdulrahmanHadz** [comment]
For the grid view:
https://imgur.com/a/843wk

This example shows the grid view for chapters and the button to re-read even if the chapter has been marked as read, you can just press on the button to go back to chapter you were last on regardless of how the chapter is marked.

For range select this is a good example:
https://imgur.com/a/NPGKu

Being able to use range select to select and number of chapters between two points using grid view.

**Comment by Safrone** [comment]
Came to make a feature request for range select and found this. In my case it would be good enough to have a 'select all before/after' option (probably with the select all option in the drop-down). On the other hand, if I could hold down one finger and scroll with the other to select many instead of having to go each one, that would make life much easier.

**Comment by eji30** [comment]
Agree with range selection. Really waiting for this feature (ಥ_ಥ)

**Comment by arkon** [comment]
Closing as duplicate of #1493 for range selection.
